tests/api/test_notification.py

```diff
@@ -0,0 +1,162 @@
+        conn.create_topic(Name="CostUsageReportUpdateTopic")
+        response = conn.list_topics()
+        topic_arn = response["Topics"][0]['TopicArn']
+
+        # Subscribe to topic
+        subscribe_response = conn.subscribe(TopicArn=topic_arn, Protocol='http', Endpoint='http://www.masu.com/api/v1/notification/')
+
+        header = {"X-Amz-Sns-Message-Type": "Notification",
+                  "X-Amz-Sns-Message-Id": "156c18c0-c49f-5067-b3ab-4b77cae67e77",
+                  "Content-Length": "1867",
+                  "Content-Type": "text/plain; charset=UTF-8",
+                  "Host": "masu.net:6868",
+                  "Connection": "Keep-Alive",
+                  "User-Agent": "Amazon Simple Notification Service Agent",
+                  "Accept-Encoding": "gzip,deflate"}
+        header['X-Amz-Sns-Topic-Arn'] = topic_arn
+        header['X-Amz-Sns-Subscription-Arn'] = subscribe_response['SubscriptionArn']
+
+        body_dict = {"Type" : "Notification",
+                     "MessageId" : "156c18c0-c49f-5067-b3ab-4b77cae67e77",
+                     "TopicArn" : topic_arn,
+                     "Subject" : "Amazon S3 Notification",
+                     "Message" : "{\"Records\":[{\"eventVersion\":\"2.0\",\"eventSource\":\"aws:s3\",\"awsRegion\":\"us-east-1\",\"eventTime\":\"2018-07-04T18:04:41.770Z\",\"eventName\":\"ObjectCreated:Put\",\"userIdentity\":{\"principalId\":\"AWS:AIDAJLSI7DTYBUZAOKHVM\"},\"requestParameters\":{\"sourceIPAddress\":\"10.88.30.162\"},\"responseElements\":{\"x-amz-request-id\":\"E7B755F7341A5745\",\"x-amz-id-2\":\"UKqyjaYspHsPnfk2S57/AtAUMqd8VO1lfuVYNP70KPfDj4ElgpulHkBBoil2PYyEdpfnLWyoUdY=\"},\"s3\":{\"s3SchemaVersion\":\"1.0\",\"configurationId\":\"CostUsageNotification\",\"bucket\":{\"name\":\"cost-usage-bucket\",\"ownerIdentity\":{\"principalId\":\"A2V75OLLZ1ABF7\"},\"arn\":\"arn:aws:s3:::cost-usage-bucket\"},\"object\":{\"key\":\"/koku/20180701-20180801/koku-Manifest.json\",\"size\":6518,\"eTag\":\"963195a6fa85a9b98a1f93a981a805fd\",\"sequencer\":\"005B3D0C39955A17E5\"}}}]}",
+                     "Timestamp" : "2018-07-04T18:04:41.844Z",
+                     "SignatureVersion" : "1",
+                     "Signature" : "J7rG3Jd9SR0So5kbkOHgfSJvOVckW5paIEc7MTTraVe1DxKzImpwHAabEAqj1dI9daPe8SlIEK92HocNjgIj9Ox3n/KE+aN871QSQ5ISG1P54uqnkaw+fwfjsFwVGnUsNQ50ntaM9JlsK6VarwnBKdEjP/EAIS00R+bueJQ0Yws8Xhvyzmw0zASRminho5EpMNpucUAY4oKjgn3Bir7MU4d/d31ZZedrdKb35XcKpIs1lZj/MPHOWRM2NDG+6AhSlEbZ94IIi9ycqoRgDn16hIBVLMbFRSdbeRjQqYUxru3Inp5upRqOUTw0yNPLPbdUD/+qiUYs9A1Mc+VPZ/M0PA==",
+                     "SigningCertURL" : "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-eaea6120e66ea12e88dcd8bcbddca752.pem",
+                     "UnsubscribeURL" : "https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:589179999999:CostUsageReportUpdateTopic:7d549a1b-502f-47f7-9469-9ed18671e76a"}
+        body = json.dumps(body_dict)
```

Comment:
It would be good to add some fuzziness to this test data. e.g. use Faker to generate the message timestamp and uuids, randomly pick regions, etc.

Not critical, but definitely nice to help improve the test quality.